doc/Customization-guide/Code-style/C-C++/index.md

```diff
@@ -0,0 +1,204 @@
+---
+layout: page
+title: C/C++
+permalink: /:path/
+sort_idx: 30
+---
+
+* TOC
+{:toc}
+
+The guide describes code formatting and coding rules that must be followed in Kaa C and C++ SDKs.
+
+There are few common and most important guidelines for both C and C++ SDK.
+
+- Use C99 for C code and C++11 for C++ code
+- Anything that causes the undefined behavior (according to C/C++ standarts) either in runtime or in compile-time is prohibited, even if at this point of time it is predictable.
+E.g. signed integer overflow.
+- Anything that relies on assumptions about unspecified behavior (according to C/C++ standarts) either in runtime or in compile-time is prohibited as well.
+E.g. assuming that function arguments evaluate in particular order.
+- Anything that relies on particular implementation-defined values, or semantically behaves differentely in different implementations should be marked and properly documented and allowed only in platform-dependent code of the Kaa SDK.
+E.g. assuming that `int` will occupy 4 bytes on every platform.
+
+  ```c
+  int *x = malloc(4); // Wrong! Size of integer is implementation-defined.
+  ```
+
+# Code appearance in C
+
+- It is recommended to use a source formatting script against modified sources.
+The script is placed in `client/client-multi/client-c/scripts/srcformat.sh`.
+- Use 4 spaces for indentation. Don't use tabs.
+- Use 100-character column width.
+- Use `snake_case` for the names (not `camelCase`) in C code.
+- All macro names must be in UPPERCASE.
+- Use [1TBS](https://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS) (the one true brace style).
+
+  ```cpp
+  int main(int argc, char *argv[])
+  {
+     if (argc > 2) {
+         printf("too many args\n");
+     } else {
+         printf("too little args\n");
+     }
+
+     return 0;
+  }
+  ```
+
+  Note brace position.
+
+- The braces in control structures are required even for a single statement.
+- The pointer star must be aligned to the variable name.
+
+  ```cpp
+  int *x;
+  ```
+
+- Labels must be indented one level less than the normal indentation (except for case labels).
+
+  ```cpp
+  int main(int argc, char *argv[])
+  {
+      switch (argc) {
+          case 3:
+              printf("hi\n");
```

Comment:
This code looks a little bit confusing. I suppose we should not go to the default statement always. 
Please, add `break` here.